lib/active_record/connection_adapters/nulldb_adapter/core.rb

```diff
@@ -343,13 +343,37 @@ def initialize_args
     [nil, @logger, @config]
   end
 
+  # Register types only once to avoid ActiveRecord::TypeConflictError
+  # in ActiveRecord::Type::Registration#<=>
+  REGISTRATION_MUTEX = Mutex.new
+
   def register_types
-    require 'active_model/type'
+    REGISTRATION_MUTEX.synchronize do
```

Comment:
`#register_types` is called whenever the adapter is initialized, which happens 2x in the specs.

the other way to avoid a TypeConflictError would be to dissolve the method and register the types when the class is loaded as it is done for other adapters.

i'm not sure the method is needed, maybe it is to avoid "polluting" the registries when the adapter isn't actually used?